src/lucky/html_builder.cr

```diff
@@ -46,7 +46,13 @@ module Lucky::HTMLBuilder
   macro generate_needy_initializer
     {% if !@type.abstract? %}
       {% sorted_assigns = ASSIGNS.sort_by { |dec|
-           dec.type.resolve.nilable? || dec.value || dec.value == nil || dec.value == false ? 1 : 0
+           has_explicit_value =
+             dec.type.is_a?(Metaclass) ||
+               dec.type.types.map(&.id).includes?(Nil.id) ||
```

Comment:
Removed resolve. This makes things a bit more reliable since it doesn't need to resolve the type at this point, it can be done later